src/Button/RedliningButton/RedliningButton.spec.jsx

```diff
@@ -0,0 +1,431 @@
+          editType: 'Edit'
+        });
+        const createSelectOrModifyInteraction = jest.spyOn(wrapper.instance(), 'createSelectOrModifyInteraction');
+
+        wrapper.instance().onToggle(true);
+        expect(createSelectOrModifyInteraction).toHaveBeenCalledTimes(1);
+
+        createSelectOrModifyInteraction.mockReset();
+        createSelectOrModifyInteraction.mockRestore();
+      });
+
+      it ('removes all draw/select interactions created by component from the map if the button was untoggled', () => {
+        const wrapper = setupWrapper();
+
+        const mockInteraction = getMockDrawPointInteraction();
+
+        const defaultMapInteractionsLength = map.getInteractions().getArray().length;
+        map.addInteraction(getMockDrawPointInteraction());
+        map.on('pointermove', wrapper.instance().onPointerMove);
+
+        wrapper.setState({
+          interaction: [mockInteraction]
+        });
+
+        expect(map.getInteractions().getArray().length).toBe(defaultMapInteractionsLength + 1);
+        expect(map.listeners_.pointermove).toBeDefined();
+
+        wrapper.instance().onToggle(false);
+
+        expect(map.listeners_.pointermove).toBeUndefined();
+      });
+
+      it ('unregisters `add` listener on redlining feature collection if drawType is Text and button was untoggled', () => {
+        const wrapper = setupWrapper();
+        wrapper.setProps({
+          drawType: 'Text'
+        });
+        const mockInteraction = getMockDrawPointInteraction();
+
+        const instance = wrapper.instance();
+        instance.createRedliningLayer();
+        instance._redliningFeatures =
+          wrapper.state().redliningLayer.getSource().getFeaturesCollection();
+
+        map.addInteraction(mockInteraction);
+        instance._redliningFeatures.on('add', wrapper.instance().handleTextAdding);
+
+        expect(instance._redliningFeatures.listeners_.add.length).toBe(2);
+
+        wrapper.setState({
+          interaction: [mockInteraction]
+        });
+
+        instance.onToggle(false);
+
+        expect(instance._redliningFeatures.listeners_.add.length).toBe(1);
+      });
+
+      it ('unregisters `select` listener on select interaction if editType is Delete and button was untoggled', () => {
+        const wrapper = setupWrapper();
+        wrapper.setProps({
+          drawType: null,
+          editType: 'Delete'
+        });
+        const mockInteraction = getMockSelectInteraction();
+
+        const instance = wrapper.instance();
+        instance._selectInteraction = mockInteraction;
+
+        map.addInteraction(mockInteraction);
+        instance._selectInteraction.on('select', wrapper.instance().onFeatureRemove);
+
+        expect(instance._selectInteraction.listeners_.select.length).toBe(1);
+
+        wrapper.setState({
+          interaction: [mockInteraction]
+        });
+
+        instance.onToggle(false);
+
+        expect(instance._selectInteraction.listeners_select).toBeUndefined();
```

Comment:
Should this bbe `instance._selectInteraction.listeners.select_`? Is this just stubbed right now?

Reply:
oh, nice catch.. fixed in 3a0a656